blog/02-15-2023-build-feature-rich-persistent-dapp-on-celo-using-wagmi/build-feature-rich-persistent-dapp-on-celo-using-wagmi.md

```diff
@@ -0,0 +1,747 @@
+      <Container maxWidth="xl">
+        <Toolbar disableGutters>
+          <Typography
+            variant="h6"
+            noWrap
+            component="a"
+            href="/"
+            sx={{
+              mr: 2,
+              display: { xs: "none", md: "flex" },
+              fontFamily: "monospace",
+              fontWeight: 700,
+              letterSpacing: ".3rem",
+              color: "purple",
+              textDecoration: "none"
+            }}
+          >
+            <span style={{ fontSize: "35px" }}>Swap</span>Lab
+          </Typography>
+          <Typography
+            variant="h5"
+            noWrap
+            component="a"
+            href=""
+            sx={{
+              mr: 2,
+              display: { xs: "flex", md: "none" },
+              flexGrow: 1,
+              fontFamily: "monospace",
+              fontWeight: 700,
+              letterSpacing: ".3rem",
+              color: "purple",
+              textDecoration: "none"
+            }}
+          >
+            <span style={{ fontSize: "35px" }}>D</span>Rythm
+          </Typography>
+          <Box sx={{ flexGrow: 1, display: "flex", justifyContent: "space-around", alignItems: 'center' }}>
+            <Typography variant="body1" sx={{color: 'purple', fontWeight: 'bolder'}}>{`${data?.formatted? data.formatted : '0.00'} $Celo`}</Typography>
+            <ConnectKitButton />
+          </Box>
+        </Toolbar>
+      </Container>
+    </AppBar>
+  );
+}
+
+export default Header;
+```
+
+For example, I displayed the balance of the connected account in the header to showcase the usage of the `useAccount` and `useBalance` hooks through property destructuring. The 'data' property returned by 'useBalance' is also an object with information regarding the balance of a connected account. From the object, we accessed _data?.formatted_ - an already formatted value of _data?.value_ which is in bigNumber format. Much of the conversion task is done for us. 
+
+The output of the 'ConnectKitButton' is displayed at the top right side of the dapp.
+![header](images/7.png)
+
+Connect wallet.
+![header](images/8.png)
+
+
+- `components/App/index.tsx`
+  - _loading_ determines when the 'Spinner' component should render.
+  - _amount_ represents the amount to approve by the users. It will be extracted from users using the `TextField` component.
+  - _value_ represents an amount user wishes to send along in a call.
+  - _allowance_: We'll fetch the user's allowance and store it here.
+  - _balance_: User's native asset's balance.
+  - _Records error message from the most recent transaction.
+  - Next, we call the 'useAccount()' hook to get the user's connected account address and the selected provider. 
+
+  >Note: The connector property is an object that represents the user's selected wallet. We use `connector?.getProvider()` to access the provider object. We will need the provider to connect to Alfajores and interact with deployed contracts.
```

Comment:
>Note: The connector property is an object that represents the user's selected wallet. We use `connector?.getProvider()` to access the provider object. The provider will need to connect to Alfajores and interact with deployed contracts.